frontend/lib/requests/lesson.ts

```
@@ -37,3 +45,273 @@ export async function getLessonBySlug<T extends Partial<Lesson> = Lesson>(
+
+export async function updateLesson(
+  id: number,
+  data: Partial<z.infer<typeof LessonSchema>>,
+  options: { reload?: boolean; regenerateSlug?: boolean } = {
+    reload: false,
+    regenerateSlug: false,
+  },
+) {
+  try {
+    if (data.blocks) {
+      data.blocks = data.blocks.map(({ id, ...rest }) => rest);
```

ESLint

[error] 'id' is defined but never used.

(@typescript-eslint/no-unused-vars)